lms/templates/courseware/progress_graph.js

```diff
@@ -128,40 +212,117 @@ $(function () {
-    legend: {show: false},
+    series: {
+        stack: true,
+        lines: {
+            show: false,
+            steps: false
+        },
+        bars: {
+            show: true,
+            barWidth: 0.8,
+            align: 'center',
+            lineWidth: 0,
+            fill: .8
+        }
+    },
+    xaxis: {
+        tickLength: 0,
+        min: 0.0,
+        max: ${tickIndex - sectionSpacer | n, dump_js_escaped_json},
+        ticks: function() {
+            for (var i = 0; i < ticks.length; i++) {
+                var tickLabel = edx.HtmlUtils.joinHtml(
+                    // The very last tick will be for the total, and it usually is composed of a number of different
+                    // grading types. To help clarify, do NOT make the label ("Total") aria-hidden in that case.
+                    edx.HtmlUtils.HTML(i < ticks.length - 1 ? '<span aria-hidden="true">' : '<span>'),
+                    ticks[i][1],
+                    edx.HtmlUtils.HTML('</span>')
+                );
+                var elementTooltips = yAxisTooltips[ticks[i][0]];
+                if (elementTooltips) {
+                    for (var tooltipIndex = 0; tooltipIndex < elementTooltips.length; tooltipIndex++) {
+                        tickLabel = edx.HtmlUtils.joinHtml(
+                            tickLabel,
+                            edx.HtmlUtils.HTML('<span class="sr">'),
+                            elementTooltips[tooltipIndex],
+                            edx.HtmlUtils.HTML('<br></span>')
+                        )
+                    }
+                }
+                ticks[i][1] = tickLabel;
+            }
+            return ticks;
+        },
+        labelAngle: 90
+    },
+    yaxis: {
+        ticks: grade_cutoff_ticks,
+        min: 0.0,
+        max: 1.0,
+        labelWidth: 100
+    },
+    grid: {
+        hoverable: true,
+        clickable: true,
+        borderWidth: 1,
+        markings: markings
+    },
+    legend: {
+        show: false
+    }
   };
   
-  var $grade_detail_graph = $("#${graph_div_id}");
+  var $grade_detail_graph = $("#${graph_div_id | n, js_escaped_string}");
   if ($grade_detail_graph.length > 0) {
     var plot = $.plot($grade_detail_graph, series, options);
     
     %if show_grade_breakdown:
-      var o = plot.pointOffset({x: ${overviewBarX} , y: ${totalScore}});
-      $grade_detail_graph.append('<div style="position:absolute;left:' + (o.left - 12) + 'px;top:' + (o.top - 20) + 'px">${"{totalscore:.0%}".format(totalscore=totalScore)}</div>');
+      var o = plot.pointOffset(
+          {x: ${overviewBarX | n, dump_js_escaped_json} , y: ${totalScore | n, dump_js_escaped_json}}
+      );
+
+      edx.HtmlUtils.append(
+          $grade_detail_graph,
+          edx.HtmlUtils.joinHtml(
+              // safe-lint: disable=javascript-concat-html
+              edx.HtmlUtils.HTML('<div class="overallGrade" style="position:absolute;left:' + (o.left - 12) + 'px;top:' + (o.top - 20) + 'px">'),
```

Comment:
Oh, ok, in that case:

```javascript
edx.HtmlUtils.HTML(
    edx.StringUtils.interpolate(
        '<div class="overallGrade" style="position:absolute;left:{offsetLeft}px;top:{offsetTop}px">'),
        {
            offsetLeft: o.left - 12,
            offsetTop: o.top - 20
        }
    )
),
```

(Yes, I'm being a little pedantic.... obviously this is all pretty poor quality code (as demonstrated by the fact that the linter won't even _run_ on it) so a string concatenation here is not the worst of the evils in this file... but we do have tools for this and we should use them if we're planning on keeping this code maintainable. But if we're going to throw it all out and rewrite in the tech debt story you mentioned, whatever works.)